src/dbnode/persist/fs/volume.go

```diff
@@ -0,0 +1,51 @@
+package fs
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/m3db/m3/src/x/ident"
+)
+
+// GetVolumeIndices returns available volume indices for specified options.
+func GetVolumeIndices(
+	filePathPrefix string,
+	namespace ident.ID,
+	shard uint32,
+	blockStart time.Time,
+) ([]int, error) {
+	shardDir := ShardDataDirPath(filePathPrefix, namespace, shard)
+	// Get volume indices based on info files.
+	r := regexp.MustCompile(fmt.Sprintf(
+		`%s/%s%s%d%s(?P<VolumeIndex>\d+)%s%s%s`,
+		shardDir,
+		filesetFilePrefix,
+		separator,
+		blockStart.UnixNano(),
+		separator,
+		separator,
+		infoFileSuffix,
+		fileSuffix,
+	))
```

Comment:
Perhaps just use "regexp.Compile(..)" and deal with the error by returning if it fails to compile (due to some of the input being unexpected?).